### PR TITLE
Add test for BZ-1869865

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### **Test Result:**
**Satellite:**
```
$ pytest -v --ansible-host-pattern server --ansible-user=root  --ansible-inventory testfm/inventory tests/test_health.py -k test_positive_check_non_rh_packages
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-3.6.1, py-1.10.0, pluggy-0.6.0 -- /home/gtalreja/sat_workspace/testfm/.testfm/bin/python3.8
cachedir: .pytest_cache
ansible: 2.9.20
rootdir: /home/gtalreja/sat_workspace/testfm, inifile:
plugins: ansible-2.2.4
collected 27 items / 26 deselected                                                                                                                

tests/test_health.py::test_positive_check_non_rh_packages PASSED                                                                            [100%]

==================================================== 1 passed, 26 deselected in 150.10 seconds ====================================================

```
**Capsule**:
```
$ pytest -v --ansible-host-pattern server --ansible-user=root  --ansible-inventory testfm/inventory tests/test_health.py -k test_positive_check_non_rh_packages
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-3.6.1, py-1.10.0, pluggy-0.6.0 -- /home/gtalreja/sat_workspace/testfm/.testfm/bin/python3.8
cachedir: .pytest_cache
ansible: 2.9.20
rootdir: /home/gtalreja/sat_workspace/testfm, inifile:
plugins: ansible-2.2.4
collected 27 items / 26 deselected                                                                                                                

tests/test_health.py::test_positive_check_non_rh_packages PASSED                                                                            [100%]

==================================================== 1 passed, 26 deselected in 135.64 seconds ====================================================

```
Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

File: tests/test_health.py [hunk 1]
```diff
@@ -827,3 +827,32 @@ def test_positive_remove_job_file(setup_subscribe_to_cdn_dogfood, ansible_module
...
+
+    :id: 2b7a25bb-902c-4a87-bb71-f460dcce655c
+
+    :setup:
+        1. foreman-maintain should be installed.
+        2. Configure custom repository and install custom package
+
+    :steps:
+        1. Run foreman-maintain health check --label non-rh-packages.
+        3. Verify presense of non-RH/custom package installed on system.
```
Comment (jameerpathan111) [inline tests/test_health.py:844]:
```suggestion
        2. Verify presense of non-RH/custom package installed on system.
```

File: tests/conftest.py [hunk 2]
```diff
@@ -506,3 +506,40 @@ def teardown_yum_content():
...
+        baseurl=f"{FAKE_YUM0_REPO}",
+        enabled="yes",
+        gpgcheck="no",
+    )
+    setup = ansible_module.file(path="/etc/yum.repos.d/custom_repo.repo", state="present")
+    assert setup.values()[0]["changed"] == 0
+
+    pkgs_locked = ansible_module.command(Packages.is_locked()).values()[0]["rc"]
+    if pkgs_locked == 0:
+        ansible_module.command(Packages.unlock())
```
Comment (jameerpathan111) [inline tests/conftest.py:527]: @Gauravtalreja1 basically you are running this test when FM Packages lock is disabled. I don't think this is right way to test this. You should enable package lock after installing the `walrus` package.

File: tests/test_health.py [hunk 3]
```diff
@@ -827,3 +827,32 @@ def test_positive_remove_job_file(setup_subscribe_to_cdn_dogfood, ansible_module
...
+
+    :expectedresults: check-non-rh-packages should work.
+
+    :CaseImportance: High
+    """
+    contacted = ansible_module.command(Health.check({"label": "non-rh-packages"}))
+    for result in contacted.values():
+        logger.info(result["stdout"])
+        assert "Found 1 unexpected non Red Hat Package(s) installed!" in result["stdout"]
+        assert "walrus-5.21-1.noarch" in result["stdout"]
```
Comment (jameerpathan111) [inline tests/test_health.py:856]: @Gauravtalreja1 will this `walrus` package version will stay same in near future? or should we change this assertion as `assert "walrus" in result["stdout"]`?

Comment (Gauravtalreja1) [reply tests/test_health.py:856]: There is a rare possibility of changing versions for packages under custom repos, but if you insist I can change this assert as you mentioned

Comment (jameerpathan111) [reply tests/test_health.py:856]: no need then.

File: tests/conftest.py (hunk 2 above)
Comment (Gauravtalreja1) [reply tests/conftest.py:550]: @jameerpathan111 Updated. request to re-review.

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**